Boss/Mod/ConstructedListpeers.cpp

```diff
@@ -31,17 +33,25 @@ Boss::Mod::ConstructedListpeers convert_legacy_listpeers(Jsmn::Object const & le
 	return cpeers;
 }
 
-std::ostream& operator<<(std::ostream& os, Boss::Mod::ConstructedListpeers const& o) {
-	for (auto p : o) {
-		os << p.first << ':'
-                   << "connected: " << p.second.connected
-                   << ", channels: ";
-                for (auto c : p.second.channels) {
-                  os << c;
-                }
-        }
+std::ostream& operator<<(std::ostream& os, ConstructedListpeer const& o) {
+	os << "(ConstructedListpeer): {"
+	   << "connected: " << o.connected
+	   << ", channels: ";
+	Util::stream_elements(os, o.channels);
```

Comment:
`listpeers` should not have the channels anymore, so we may want drop it?

Reply:
So this is a `ConstructedListpeer` which is a "compatibility struct" invented by me to make the conversion easier.  Basically I convert the return from `listpeerchannels` into what the old `listpeers` output used to look like, so the consumers do not need to invert their loops and logic.

Reply:
> wondering if this is due some channels that are null?

I think so too, specifically I'm guessing that they have channels which don't have `status` fields.  The code look for `status` w/o checking to see if the field exists.

Based on prior experience these poorly formed channels might exist early in their lifecycle, before they are confirmed.

I expected to see the errors during initial channel creation and didn't in my most recent run, so there is some chance I don't understand the mechanism completetly though.

Reply:
Mh these are good point, but the status should always exist otherwise there is a bug in cln, I will try to look inside the codebase in this week